File: libs/python_common/src/nodename.py
from packed_data import MaxLengthCString, ValidationError

class NodeName(MaxLengthCString):
    NODE_NAME_SIZE = 32

    def __init__(self, name="unnamed"):
        super(NodeName, self).__init__(NodeName.NODE_NAME_SIZE)
        self.name = name
        
    def name():
        doc = """Property for name."""
        def fget(self):
            return self.string
        def fset(self, value):
            if len(value) == 0:
                raise ValidationError("A node name cannot be the empty string.")
            self.string = value
        return locals()
    name = property(**name())

File: daemon/qPresenceCommon/nodename.cpp
#include "nodename.h"

NodeName::NodeName(const QByteArray &name) : MaxLengthCString(NODE_NAME_SIZE)
{
    setName(name);
}

NodeName::NodeName(const NodeName &other) : MaxLengthCString(NODE_NAME_SIZE)
{
    this->setName(other.name());
}

NodeName& NodeName::operator=(const NodeName &other)
{
    this->setName(other.name());
    return *this;
}

bool NodeName::setName(const QByteArray &name)
{
    return setStr(name);
}

const QByteArray& NodeName::name() const
{
    return str();
}
